core/lib/dal/src/applied_snapshot_status_dal.rs

```diff
@@ -0,0 +1,142 @@
+use zksync_types::{snapshots::AppliedSnapshotStatus, L1BatchNumber, MiniblockNumber};
+
+use crate::StorageProcessor;
+
+#[derive(Debug)]
+pub struct AppliedSnapshotStatusDal<'a, 'c> {
+    pub(crate) storage: &'a mut StorageProcessor<'c>,
+}
+
+impl AppliedSnapshotStatusDal<'_, '_> {
+    pub async fn set_applied_snapshot_status(
```

Comment:
This can be done later, but I think it makes sense in general to have a more typesafe interface by having separate methods for creating and updating the snapshot:

```rust
pub async fn create_snapshot_recovery_status(
    &mut self,
    status: &SnapshotRecoveryStatus
) -> sqlx::Result<()> {
    // Uses `INSERT`; fails if the `SnapshotRecoveryStatus` exists
}

pub async fn set_last_finished_chunk_id(
    &mut self,
    l1_batch_number: L1BatchNumber,
    last_finished_chunk_id: u64,
) -> sqlx::Result<()> {
    // Uses `UPDATE`; fails if the snapshot doesn't exist
}
```

This is how it's done in other DALs (e.g., see L1 batch lifecycle in `BlocksDal`), and IMO it makes code easier to maintain in general. If we don't have full understanding of modifying methods, we can skip adding them for now.

Reply:
I agree it may make sense to refactor this in the future, but after this table "stabilizes"